src/renderer/mesh_loader.zig

```diff
@@ -34,12 +35,61 @@ pub fn loadObjMeshFromFile(
     var uvs = std.ArrayList([2]f32).init(arena);
 
     var buf: [1024]u8 = undefined;
+    var inside_object: bool = false;
+    var previous_obj_positions_count: u32 = 0;
+    var previous_obj_uvs_count: u32 = 0;
+    var previous_obj_normals_count: u32 = 0;
+
+    var mesh = Mesh{
+        .num_lods = 0,
+        .lods = undefined,
+    };
 
     while (try in_stream.readUntilDelimiterOrEof(&buf, '\n')) |line| {
         var it = std.mem.split(u8, line, " ");
 
         var first = it.first();
-        if (std.mem.eql(u8, first, "v")) {
+        if (std.mem.eql(u8, first, "o")) {
+            if (inside_object) {
+                try storeMeshLod(
```

Comment:
In general I don't think we need to use error handling for catching out-of-memory errors (which I think this is?). I'm not strictly opposed to it, especially if it's code that might be upstreamed to zig-gamedev or if it feels "library-ish", but I'm only aiming for a PC version i.e. effectively infinite memory. 

So if we can make the code simpler in that respect I'm for it :)

Reply:
Not sure if it's clear here but I was referring to the `try` :)

Reply:
haha yes, it was clear :)
I should replace it with `catch unreachable`, right?

Reply:
If storeMeshLod is "our" function I would just move any catch unreachable in there and make it not return an error. But I'm not really opposed to it being more proper and doing memory error handling, I just don't do it when I'm coding since I don't think that will be an issue. Soooo... tbh you can just leave it the way it is :D